### PR TITLE
Implement composite, analytically divergence-free B-spline kernels

### PR DESCRIPTION
**Pull request by boyceg**
This is intended to provide the noninterpolatory divergence-free kernels from https://doi.org/10.1016/j.jcp.2022.111500.

### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

**Comment by boyceg** [comment]
This is intended to provide the noninterpolatory divergence-free kernels from https://doi.org/10.1016/j.jcp.2022.111500.

**Comment by boyceg** [comment]
@colegruninger97 could you take a look and see if it corresponds to your MATLAB implementation?

**Comment by drwells** [comment]
Adding tests should be pretty easy - just duplicate input files for `interpolate_velocity_01_2d` (and make sure we get good max norm errors). We don't really have 'manufactured solutions' tests with spreading - there we just print values.

File: ibtk/src/lagrangian/LEInteractor.cpp [hunk 1]
```diff
@@ -1367,6 +1513,18 @@ string_to_kernel(const std::string& kernel_fcn)
...
+    {
+        switch (kernel_fcn.back())
+        {
+        case '2':
+            return COMPOSITE_BSPLINE_32;
+        case '3':
+            return COMPOSITE_BSPLINE_43;
+        }
+        break;
+    }
```
Comment (drwells) [inline ibtk/src/lagrangian/LEInteractor.cpp:1527]: ✔️

File: ibtk/src/lagrangian/fortran/lagrangian_interaction2d.f.m4 [hunk 2]
```diff
@@ -3187,3 +3187,486 @@ c
...
+     &                 (X(d,s)+Xshift(d,l)-X_cell(d))/dx(d))
+               endif
+            enddo
+         enddo
+c
+c     Interpolate u onto V.
+c
+         INTERPOLATE_2D_SPECIALIZE_FIXED_WIDTH(ic_lower(1), ic_upper(1),
+                                               ic_lower(0), ic_upper(0),
+                                               3)
```
Comment (drwells) [inline ibtk/src/lagrangian/fortran/lagrangian_interaction2d.f.m4:3298]: if we are reading a 3x3 stencil here then we need to populate the extra rows (or columns) of `w` with zeros.

Comment (boyceg) [reply ibtk/src/lagrangian/fortran/lagrangian_interaction2d.f.m4:3298]: That happens automatically because the basic kernel functions all return 0 outside of their radius of support.

Comment (boyceg) [reply ibtk/src/lagrangian/fortran/lagrangian_interaction2d.f.m4:3298]: However, per #1501, we could probably improve performance by making specialized helper macros that only use the nonzero values. For staggered data, we also probably also could relax the ghost cell width requirements for these kernels.

**Comment by colegruninger97** [comment]
Okay, after a bit of foiling, I can confirm that you are using the same Bspline combinations as https://www.sciencedirect.com/science/article/pii/S0021999122005629?via%3Dihub

**Comment by boyceg** [comment]
Great. Do you have code for the interpolatory versions of those kernels that could be added for comparison?

**Comment by colegruninger97** [comment]
> Great. Do you have code for the interpolatory versions of those kernels that could be added for comparison?

I haven't yet tried implementing the interpolatory verions of the kernels; however, I'd be happy to give it a shot.

**Comment by boyceg** [comment]
I think this is ready to merge.

**Comment by boyceg** [comment]
@drwells I think this is good to merge. We still need to investigate the performance of these kernels, but that seems outside the scope of this PR. 😁

**Comment by boyceg** [comment]
Whoops, just found a Fortran interface error.

**Comment by boyceg** [comment]
OK, fixed now!

File: tests/spread/spread_01_2d.composite_bspline_43.mpirun=4.output [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+[(8,4),(11,7)]
+[(12,0),(15,3)]
+[(12,4),(15,7)]
+rank: 3 level: 1 boxes:
+[(8,8),(11,11)]
+[(8,12),(11,15)]
+[(12,8),(15,11)]
+[(12,12),(15,15)]
+
+rank: 0
```
Comment (drwells) [inline tests/spread/spread_01_2d.composite_bspline_43.mpirun=4.output:29]: I think N needs to be bigger to get some values printed here.

Comment (boyceg) [reply tests/spread/spread_01_2d.composite_bspline_43.mpirun=4.output:29]: Hrumm.

Comment (boyceg) [reply tests/spread/spread_01_2d.composite_bspline_43.mpirun=4.output:29]: Damn Fortran. I lopped off the last argument of the composite B-spline 4/3 spreading subroutine.

Comment (boyceg) [reply tests/spread/spread_01_2d.composite_bspline_43.mpirun=4.output:29]: Take another look please. 😁

**Comment by boyceg** [comment]
@drwells, I fixed the error that was resulting in empty output for the `COMPOSITE_BSPLINE_43` spreading test.
